Lib/subprocess.py

```diff
@@ -409,8 +409,9 @@ def check_output(*popenargs, timeout=None, **kwargs):
     decoded according to locale encoding, or by "encoding" if set. Text mode
     is triggered by setting any of text, encoding, errors or universal_newlines.
     """
-    if 'stdout' in kwargs:
-        raise ValueError('stdout argument not allowed, it will be overridden.')
+    for kw in ('stdout', 'check'):
```

Comment:
rather than completely forbidding passing of `check` lets remain compatible with existing behavior but ban it if it isn't truthy.

```
if not kwargs.get('check', True):
    raise ValueError(''The "check" arg is meaningless here, non-True values not allowed.')
```

That way any existing code that already passes `check=True` keeps working, we don't need a behavior deprecation period, and the error message for existing code that already fails today with the odd looking TypeError as described in the bug now has more appropriate ValueError and nice error message.

This also needs to be added to `check_call()` above.

Reply:
Oh hah. my bad, no need for this.  it's already an error even if check=True.  proceed with your existing code. :)